pkgs/os-specific/linux/tuxedo-keyboard/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, kernel, kmod }:
+
+stdenv.mkDerivation rec {
+  name = "tuxedo-keyboard-${kernel.version}-${version}";
+  version = "2019-08-26";
+
+  src = fetchFromGitHub {
+    owner = "tuxedocomputers";
+    repo = "tuxedo-keyboard";
+    rev = "d65e76e84cfd8169591fc2a0a7c9219fa19da1b5";
+    sha256 = "1s48qpwybwh5pwqas2d1v2a7x4r97sm4hr9i4902r1d7h384bv17";
+  };
+
+  preConfigure = ''
+    substituteInPlace Makefile --replace '/lib/modules/' "${kernel.dev}/lib/modules/"
+    substituteInPlace Makefile --replace '$(shell uname -r)' "${kernel.modDirVersion}"
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/lib/modules/${kernel.modDirVersion}"
+    mv src/tuxedo_keyboard.ko $out/lib/modules/${kernel.modDirVersion}
```

Comment:
cc @zimbatm Sorry to bother you again :) Do you have an idea how this could be solved more elegantly? I'm a bit stuck here again.